pilot/pkg/xds/debug.go

```diff
@@ -290,6 +290,9 @@ func (s *DiscoveryServer) Syncz(w http.ResponseWriter, req *http.Request) {
 			})
 		}
 	}
+	sort.Slice(syncz, func(i, j int) bool {
```

Comment:
can you sort this namespace first then name secondary

Reply:
This is a good advice

Reply:
Why do we sort? I don't understand the intent of this PR, and if I saw the code I would not understand either.

Can we add a comment in the code explaining why?

Reply:
The sort is to make the output have a determined order, this is the only benefit. 

Since it has no performance effect on main path, lgtm

Reply:
When I query the proxys information, I noticed that the output list of proxies is always unordered. I thinke that having the list sorted in a specific order can improve user experience and make it more user-friendly.